recipes/dbus/1.x.x/conanfile.py

```diff
@@ -0,0 +1,108 @@
+
+    options = {
+        "fPIC": [True, False],
+        "with_x11": [True, False],
+        "with_glib": [True, False],
+        "enable_assert": [True, False],
+        "enable_checks": [True, False],
+        "install_system_libs": [True, False]}
+
+    default_options = {
+        "fPIC": True,
+        "with_x11": False,
+        "with_glib": False,
+        "enable_assert": True,
+        "enable_checks": True,
+        "install_system_libs": False}
+
+    generators = "cmake", "cmake_find_package", "cmake_paths"
+
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+    _cmake = None
+
+    requires = (
+        "expat/2.2.9"
+    )
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            raise ConanInvalidConfiguration("D-Bus is not compatible with Windows")
+        if self.settings.os == "Macos":
+            raise ConanInvalidConfiguration("D-Bus is not compatible with MacOS")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def requirements(self):
+        if self.options.with_glib:
+            self.requires("glib/2.65.0")
+
+        if self.options.with_x11:
+            self.requires("xorg/system")
+
+    def _configure_cmake(self):
+        if not self._cmake:
+            self._cmake = CMake(self)
+
+            self._cmake.definitions["DBUS_BUILD_TESTS"] = False
+            self._cmake.definitions["DBUS_ENABLE_DOXYGEN_DOCS"] = False
+            self._cmake.definitions["DBUS_ENABLE_XML_DOCS"] = False
+
+            self._cmake.definitions["DBUS_BUILD_X11"] = self.options.with_x11
+            self._cmake.definitions["DBUS_WITH_GLIB"] = self.options.with_glib
+            self._cmake.definitions["DBUS_DISABLE_ASSERT"] = not self.options.enable_assert
+            self._cmake.definitions["DBUS_DISABLE_CHECKS"] = not self.options.enable_checks
+            self._cmake.definitions["DBUS_INSTALL_SYSTEM_LIBS"] = self.options.install_system_libs
+
+            path_to_cmake_lists = os.path.join(self._source_subfolder, "cmake")
+
+            self._cmake.configure(source_folder=path_to_cmake_lists,
+                                  build_folder=self._build_subfolder)
+        return self._cmake
+
+    def build(self):
+        dbus_cmake = tools.os.path.join(
+            self._source_subfolder, "cmake", "CMakeLists.txt")
+
+        if self.options.with_glib:
+            tools.replace_in_file(dbus_cmake, "GLib2", "glib")
+            tools.replace_in_file(dbus_cmake, "GLIB2", "GLIB")
+
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy(pattern="COPYING", dst="licenses",
+                  src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        for i in ["var", "share", "bin", "etc"]:
```

Comment:
why removing the `bin` folder ? it is allowed in CCI packages
```suggestion
        for i in ["var", "share", "etc"]:
```

Reply:
D-Bus generate `dbus-daemon`, `dbus-launch` and other execuable files in `bin` folder, but for some reasons executables linking at the system paths like `/lib/x86_64-linux-gnu/libdbus-1.so.3`, maybe it's dbus bug or I can how to change this behavior in conan or cmake, but I don't know how?!

Reply:
isn't this just a problem of `LD_LIBRARY_PATH`?